### DISCUSSION TITLE
Change the look of <DragOverlay> as it is snapping back from being released

### DISCUSSION THREAD

**Discussion by pstrassmann** [clauderic/dnd-kit#762] [Q&A]
When dragging a component and releasing it over a non-droppable area, the component snaps back to its original location. Is there a way to change the rendering of `<DragOverlay>` just in the time it is "snapping back"? 

Example:

1) Click a draggable component that looks like _style 1_
2) Use a piece of state (`activeId`) to render a `<DragOverlay>` of _style 2_ when it is being dragged
3) Release the _style 2_ <DragOverlay> over a non droppable area
4) **_I'd like to:_**  make the `<DragOverlay>` switch to looking like _style 1_ (the original component look) as it is snapping back.

Any ideas of how I might do this? Thanks so much!

**Comment by aroman** [comment]
@pstrassmann did you ever solve this? running into the same issue. strangely, it seems like `DragOverlay` doesn't re-render based on the result of state changes in the `DndContext.onDragEnd` callback

**Reply by pstrassmann** [reply]
@aroman See this section of the [CHANGELOG](https://github.com/clauderic/dnd-kit/blob/master/packages/core/CHANGELOG.md#changes-to-the-dropanimation-prop) for 'sideEffects' that can be applied during the drop animation. There are a number of really cool things you can do with this feature, although I would need to do more exploration to see how this could be integrated with state changes.

For example, to make the `active` invisible and the `Drag Overlay` semi-transparent during the drop animation, you can do the following:

```tsx
const customDropAnimation: DropAnimationOptions = {
  duration: 250,
  easing: 'ease',
  keyframes: defaultDropAnimationKeyframes,
  sideEffects: defaultDropAnimationSideEffects({
    styles: {
      active: {
        opacity: '0',
      },
     dragOverlay: {
       opacity: '0.5' // set the dragOverlay opacity to 0.5 during the drop animation
     },
    },
  }),
};

<DragOverlay dropAnimation = { customDropAnimation } > ....
```


Hope this helps!

**Reply by beckaberhanu** [reply]
I haven't been able to find any documentation for how to use this sideEffects property. If it is described in more detail somewhere can you please point me to it?

**Reply by pstrassmann** [reply]
@beckaberhanu There are not docs on this that I'm aware of, other than what's in the [changelog](https://github.com/clauderic/dnd-kit/blob/master/packages/core/CHANGELOG.md#changes-to-the-dropanimation-prop). I think your best bet at the moment is going to be to dig into the implementation of useDropAnimation.ts [here](https://github.com/clauderic/dnd-kit/blob/master/packages/core/src/components/DragOverlay/hooks/useDropAnimation.ts) how to see how sideEffects are typed and used
